modules/Sources/Dependencies/AddressBookClient/AddressBookLiveKey.swift

```diff
@@ -173,60 +194,70 @@ extension AddressBookClient: DependencyKey {
 
         remoteContacts.contacts.forEach {
             var notFound = true
-            var indexToUpdate = -1
             
             for i in 0..<syncedContacts.contacts.count {
                 let contact = syncedContacts.contacts[i]
                 
                 if $0.id == contact.id {
                     notFound = false
                     
-                    if $0.lastUpdated > contact.lastUpdated {
-                        indexToUpdate = i
+                    // If the timestamps are equal, the local entry takes priority.
+                    if $0.lastUpdated >= contact.lastUpdated {
+                        syncedContacts.contacts[i].name = $0.name
+                        syncedContacts.contacts[i].lastUpdated = $0.lastUpdated
+                        shouldUpdateRemote = true
```

Comment:
This is a more serious bug: the `shouldUpdateRemote` logic absolutely depended on merging from local into remote. This way around it will be set to true only when the remote has an entry that should be updated locally, which is completely wrong for the way this variable is used.

Reply:
Actually, we determined that mutations to the address book cause an unconditional remote store, so this has much smaller impact than I originally thought (although it's still a bug relative to the intended behaviour of `syncContacts` with `storeAfterSync: true`).